crates/wmemcheck/Cargo.toml

```diff
@@ -0,0 +1,11 @@
+[package]
+name = "wmemcheck"
```

Comment:
Let's rename this to `wasmtime-wmemcheck`, to stick with the convention that everything in `crates/` is named with that prefix.